### ISSUE TITLE
Got error: Concatenation: all concat input tensors must have the same dimensions except on the concatenation axis

### ISSUE THREAD

**Issue by yuanzhedong** [onnx/onnx-tensorrt#666]
## Environment

**TensorRT Version**: 5.1
**ONNX-TensorRT Version / Branch**: 5.1
**GPU Type**: RTX2080
**Nvidia Driver Version**: 450.102.04
**CUDA Version**: 10.0
**CUDNN Version**: 
**Operating System + Version**: ubuntu 18.04
**Python Version (if applicable)**: 
**TensorFlow + TF2ONNX Version (if applicable)**: TF2.4.1 TF2ONNX:2.4.1
**PyTorch Version (if applicable)**: 
**Baremetal or Container (if container which image + tag)**: 

## Steps To Reproduce
I'm trying to convert a unet model trained with TF2 to TRT5. the TF2 to onnx seems working, I'm able to use onnxruntime to load the onnx and print some inference output. 
```
python -m tf2onnx.convert --saved-model /home/landingai/Downloads/best_model --output /home/landingai/Downloads/model.onnx --opset 9
```
But I have some trouble to covert onnx to trt5, first I applied the change in this PR:https://github.com/onnx/onnx-tensorrt/issues/199#issuecomment-524145151

Then try to convert onnx to trt:
```
(tf2onnx) landingai@tensorbook:~/landing/onnx-tensorrt$ onnx2trt ~/Downloads/model.onnx -o e.trt -v
----------------------------------------------------------------
Input filename:   /home/landingai/Downloads/model.onnx
ONNX IR version:  0.0.4
Opset version:    9
Producer name:    tf2onnx
Producer version: 1.8.4
Domain:           
Model version:    0
Doc string:       
----------------------------------------------------------------
Parsing model
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/bn_data/FusedBatchNormV3__151:0:Transpose -> (3, 512, 512)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/bn_data/FusedBatchNormV3:0:BatchNormalization -> (3, 512, 512)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/conv0/Conv2D:0:Conv -> (64, 256, 256)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/bn0/FusedBatchNormV3:0:BatchNormalization -> (64, 256, 256)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/relu0/Relu:0:Relu -> (64, 256, 256)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/zero_padding2d_1/Pad:0:Pad -> (64, 258, 258)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/pooling0/MaxPool:0:MaxPool -> (64, 128, 128)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage1_unit1_bn1/FusedBatchNormV3:0:BatchNormalization -> (64, 128, 128)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage1_unit1_relu1/Relu:0:Relu -> (64, 128, 128)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage1_unit1_sc/Conv2D:0:Conv -> (64, 128, 128)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage1_unit1_bn2/FusedBatchNormV3:0:Conv -> (64, 128, 128)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage1_unit1_relu2/Relu:0:Relu -> (64, 128, 128)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage1_unit1_conv2/Conv2D:0:Conv -> (64, 128, 128)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/add/add:0:Add -> (64, 128, 128)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage1_unit2_bn1/FusedBatchNormV3:0:BatchNormalization -> (64, 128, 128)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage1_unit2_relu1/Relu:0:Relu -> (64, 128, 128)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage1_unit2_bn2/FusedBatchNormV3:0:Conv -> (64, 128, 128)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage1_unit2_relu2/Relu:0:Relu -> (64, 128, 128)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage1_unit2_conv2/Conv2D:0:Conv -> (64, 128, 128)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/add_1/add:0:Add -> (64, 128, 128)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage1_unit3_bn1/FusedBatchNormV3:0:BatchNormalization -> (64, 128, 128)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage1_unit3_relu1/Relu:0:Relu -> (64, 128, 128)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage1_unit3_bn2/FusedBatchNormV3:0:Conv -> (64, 128, 128)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage1_unit3_relu2/Relu:0:Relu -> (64, 128, 128)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage1_unit3_conv2/Conv2D:0:Conv -> (64, 128, 128)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/add_2/add:0:Add -> (64, 128, 128)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage2_unit1_bn1/FusedBatchNormV3:0:BatchNormalization -> (64, 128, 128)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage2_unit1_relu1/Relu:0:Relu -> (64, 128, 128)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage2_unit1_sc/Conv2D:0:Conv -> (128, 64, 64)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage2_unit1_bn2/FusedBatchNormV3:0:Conv -> (128, 64, 64)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage2_unit1_relu2/Relu:0:Relu -> (128, 64, 64)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage2_unit1_conv2/Conv2D:0:Conv -> (128, 64, 64)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/add_3/add:0:Add -> (128, 64, 64)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage2_unit2_bn1/FusedBatchNormV3:0:BatchNormalization -> (128, 64, 64)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage2_unit2_relu1/Relu:0:Relu -> (128, 64, 64)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage2_unit2_bn2/FusedBatchNormV3:0:Conv -> (128, 64, 64)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage2_unit2_relu2/Relu:0:Relu -> (128, 64, 64)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage2_unit2_conv2/Conv2D:0:Conv -> (128, 64, 64)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/add_4/add:0:Add -> (128, 64, 64)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage2_unit3_bn1/FusedBatchNormV3:0:BatchNormalization -> (128, 64, 64)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage2_unit3_relu1/Relu:0:Relu -> (128, 64, 64)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage2_unit3_bn2/FusedBatchNormV3:0:Conv -> (128, 64, 64)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage2_unit3_relu2/Relu:0:Relu -> (128, 64, 64)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage2_unit3_conv2/Conv2D:0:Conv -> (128, 64, 64)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/add_5/add:0:Add -> (128, 64, 64)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage2_unit4_bn1/FusedBatchNormV3:0:BatchNormalization -> (128, 64, 64)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage2_unit4_relu1/Relu:0:Relu -> (128, 64, 64)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage2_unit4_bn2/FusedBatchNormV3:0:Conv -> (128, 64, 64)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage2_unit4_relu2/Relu:0:Relu -> (128, 64, 64)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage2_unit4_conv2/Conv2D:0:Conv -> (128, 64, 64)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/add_6/add:0:Add -> (128, 64, 64)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage3_unit1_bn1/FusedBatchNormV3:0:BatchNormalization -> (128, 64, 64)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage3_unit1_relu1/Relu:0:Relu -> (128, 64, 64)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage3_unit1_sc/Conv2D:0:Conv -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage3_unit1_bn2/FusedBatchNormV3:0:Conv -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage3_unit1_relu2/Relu:0:Relu -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage3_unit1_conv2/Conv2D:0:Conv -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/add_7/add:0:Add -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage3_unit2_bn1/FusedBatchNormV3:0:BatchNormalization -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage3_unit2_relu1/Relu:0:Relu -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage3_unit2_bn2/FusedBatchNormV3:0:Conv -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage3_unit2_relu2/Relu:0:Relu -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage3_unit2_conv2/Conv2D:0:Conv -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/add_8/add:0:Add -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage3_unit3_bn1/FusedBatchNormV3:0:BatchNormalization -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage3_unit3_relu1/Relu:0:Relu -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage3_unit3_bn2/FusedBatchNormV3:0:Conv -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage3_unit3_relu2/Relu:0:Relu -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage3_unit3_conv2/Conv2D:0:Conv -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/add_9/add:0:Add -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage3_unit4_bn1/FusedBatchNormV3:0:BatchNormalization -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage3_unit4_relu1/Relu:0:Relu -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage3_unit4_bn2/FusedBatchNormV3:0:Conv -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage3_unit4_relu2/Relu:0:Relu -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage3_unit4_conv2/Conv2D:0:Conv -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/add_10/add:0:Add -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage3_unit5_bn1/FusedBatchNormV3:0:BatchNormalization -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage3_unit5_relu1/Relu:0:Relu -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage3_unit5_bn2/FusedBatchNormV3:0:Conv -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage3_unit5_relu2/Relu:0:Relu -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage3_unit5_conv2/Conv2D:0:Conv -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/add_11/add:0:Add -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage3_unit6_bn1/FusedBatchNormV3:0:BatchNormalization -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage3_unit6_relu1/Relu:0:Relu -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage3_unit6_bn2/FusedBatchNormV3:0:Conv -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage3_unit6_relu2/Relu:0:Relu -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage3_unit6_conv2/Conv2D:0:Conv -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/add_12/add:0:Add -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage4_unit1_bn1/FusedBatchNormV3:0:BatchNormalization -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage4_unit1_relu1/Relu:0:Relu -> (256, 32, 32)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage4_unit1_sc/Conv2D:0:Conv -> (512, 16, 16)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage4_unit1_bn2/FusedBatchNormV3:0:Conv -> (512, 16, 16)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage4_unit1_relu2/Relu:0:Relu -> (512, 16, 16)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage4_unit1_conv2/Conv2D:0:Conv -> (512, 16, 16)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/add_13/add:0:Add -> (512, 16, 16)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage4_unit2_bn1/FusedBatchNormV3:0:BatchNormalization -> (512, 16, 16)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage4_unit2_relu1/Relu:0:Relu -> (512, 16, 16)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage4_unit2_bn2/FusedBatchNormV3:0:Conv -> (512, 16, 16)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage4_unit2_relu2/Relu:0:Relu -> (512, 16, 16)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage4_unit2_conv2/Conv2D:0:Conv -> (512, 16, 16)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/add_14/add:0:Add -> (512, 16, 16)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage4_unit3_bn1/FusedBatchNormV3:0:BatchNormalization -> (512, 16, 16)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage4_unit3_relu1/Relu:0:Relu -> (512, 16, 16)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage4_unit3_bn2/FusedBatchNormV3:0:Conv -> (512, 16, 16)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage4_unit3_relu2/Relu:0:Relu -> (512, 16, 16)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/stage4_unit3_conv2/Conv2D:0:Conv -> (512, 16, 16)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/add_15/add:0:Add -> (512, 16, 16)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/bn1/FusedBatchNormV3:0:BatchNormalization -> (512, 16, 16)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/relu1/Relu:0:Relu -> (512, 16, 16)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/decoder_stage0a_transpose/conv2d_transpose:0:ConvTranspose -> (256, 34, 34)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/decoder_stage0a_bn/FusedBatchNormV3:0:BatchNormalization -> (256, 34, 34)
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/decoder_stage0a_relu/Relu:0:Relu -> (256, 34, 34)
[2021-04-10 19:40:24   ERROR] (Unnamed Layer* 112) [Concatenation]: all concat input tensors must have the same dimensions except on the concatenation axis
[2021-04-10 19:40:24    INFO] StatefulPartitionedCall/model_1/decoder_stage0_concat/concat:0:Concat -> (32765)
While parsing node number 113 [Conv -> "StatefulPartitionedCall/model_1/decoder_stage0b_bn/FusedBatchNormV3:0"]:
--- Begin node ---
input: "StatefulPartitionedCall/model_1/decoder_stage0_concat/concat:0"
input: "StatefulPartitionedCall/model_1/decoder_stage0b_conv/Conv2D_weights_fused_bn"
input: "StatefulPartitionedCall/model_1/decoder_stage0b_conv/Conv2D_bias_fused_bn"
output: "StatefulPartitionedCall/model_1/decoder_stage0b_bn/FusedBatchNormV3:0"
name: "StatefulPartitionedCall/model_1/decoder_stage0b_conv/Conv2D"
op_type: "Conv"
attribute {
  name: "dilations"
  ints: 1
  ints: 1
  type: INTS
}
attribute {
  name: "strides"
  ints: 1
  ints: 1
  type: INTS
}
attribute {
  name: "kernel_shape"
  ints: 3
  ints: 3
  type: INTS
}
attribute {
  name: "pads"
  ints: 1
  ints: 1
  ints: 1
  ints: 1
  type: INTS
}

--- End node ---
ERROR: /home/landingai/landing/onnx-tensorrt/builtin_op_importers.cpp:728 In function importConv:
[8] Assertion failed: dims.nbDims == 3


```

**Comment by yuanzhedong** [comment]
Try to print out the input shape under builtin_op_importers.cpp 
```
DEFINE_BUILTIN_OP_IMPORTER(Concat) {                                                                                                                                                         |
  std::vector<nvinfer1::ITensor*> tensors;                                                                                                                                                   |
  for( auto& input : inputs ) {                                                                                                                                                              |
    std::cout << "###########" << std::endl;                                                                                                                                                 |
    std::cout << input.shape() << std::endl; 
```
and get this:
```
###########
(512, 34, 34)
###########
(256, 32, 32)
```

Seems it's trying to concat this two dims and cause that error, but throw onnx file it should be 512x32x32, not sure why it becomes 34 there:
![Screenshot from 2021-04-10 20-51-53](https://user-images.githubusercontent.com/5069709/114291743-a3b30380-9a3e-11eb-82f5-e34cf8747795.png)

**Comment by yuanzhedong** [comment]
Turns out the onnx file losts the padding attr [1, 1, 1, 1] for conv2d_transpose, 
![Screenshot from 2021-04-10 23-20-56](https://user-images.githubusercontent.com/5069709/114294436-758bee80-9a53-11eb-91d5-d33e67928429.png)

hacked to add the padding info into get_kernel_params and now I'm able to get the trt engine. Not sure if there's some way I can preserve the padding info through tf2onnx?
[fix.patch.tar.gz](https://github.com/onnx/onnx-tensorrt/files/6291267/fix.patch.tar.gz)

**Comment by kevinch-nv** [comment]
Is there a reason you're using an older version of TRT 5.1? Does this issue still persist with the latest version of TRT?

**Comment by yuanzhedong** [comment]
> Is there a reason you're using an older version of TRT 5.1? Does this issue still persist with the latest version of TRT?

Thank you for your response Kevin, the reason we're using TRT5.1 is our products are released and running on jetpack 4.2, which has trt 5.1. If we want to switch to trt 7 we need to recall and reflash all our products and flash them to jetpack 4.4+. For the models we're using right now we go with tf->uff->trt, we're trying to see if it's feasible to switch to tf->onnx->trt5.

**Comment by kevinch-nv** [comment]
Thanks for the update. We do not officially support TRT 5.1 anymore, so you may encounter issues in the tf->onnx->trt workflow that will have been fixed in later versions. Please consider updating to a later version of Jetpack if you want increased onnx->trt support.

Either way it looks like you found a work around so I'll be closing this issue.
